Documentation/applications/examples/elf/index.rst

```diff
@@ -4,9 +4,16 @@
 
 This example builds a small ELF loader test case. This includes several test
 programs under ``examples/elf`` tests. These tests are build using the relocatable
-ELF format and installed in a ROMFS file system. At run time, each program in
-the ROMFS file system is executed. Requires ``CONFIG_ELF``. Other configuration
-options:
+ELF format and installed in a configurable file system. At run time, the file system
+is mounted and each program is executed. Requires ``CONFIG_ELF``.
+Two distinct types of file systems are supported: internal (included in the NuttX binary)
+and external (not included in the binary). The external file systems need to be uploaded
+manually. For the internal file systems the sample supports the ``romfs`` and ``cromfs``
+filesystems. For the external file systems any of the provided file systems can be used
+(``vfat``, ...), but also the ``romfs`` filesystem for which the image (``romfs.img``) is
+automatically generated.
```

Comment:
Maybe include the information that these filesystem when flashed in the internal Flash cannot be modified (unless the progmem be used with it, TODO).

Reply:
I might misunderstand what you mean. When the filesystem changes and the elf file for a certain program is no longer available the elf program will just report that it cannot find a certain elf program.

Reply:
I mean even when not using romfs, the content of the flash where the file is loaded cannot be modified

Reply:
I don't think this is true. When the filesystem is not mounted it can be updated and remounted. It is possible to start any program that is included in the FS after manually mounting the FS (from nsh, when starting elf's from nsh is enabled).

Reply:
You can try put a txt file in the internal flash, mount it and then modify this file, I think it will not be based back in the flash, unless you are using progmem support